include/rdma/fi_eq.h

```diff
@@ -239,6 +239,7 @@ struct fi_cq_err_entry {
 	/* err_data is available until the next time the CQ is read */
 	void			*err_data;
 	size_t			err_data_size;
+	fi_addr_t		src_addr;
```

Comment:
There's not a great place for this comment, so putting it here.  As an option, we can modify fi_cq_readerr() to:
```
static inline ssize_t
fi_cq_readerr(struct fid_cq *cq, struct fi_cq_err_entry *buf, uint64_t flags)
{
    /* For compatibility with older providers */
    buf->src_addr = FI_ADDR_NOTAVAIL;
    return cq->ops->readerr(cq, buf, flags);
}
```
If we go through this path, the app has recompiled with the updated structure.  This ensures that the src_addr has been set to a valid value for providers that aren't updated to set it directly.

This change may only be worth doing if it allows us to avoid updating all the providers.